File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio8 nota ampliada/archives/8.py
import sys
import math

# Obtener los argumentos de la línea de comandos
num1 = float(sys.argv[1])
num2 = float(sys.argv[2])
num3 = float(sys.argv[3])
if (
    (num1 >= 0 and num1 <= 100)
    and (num2 >= 0 and num2 <= 100)
    and (num3 >= 0 and num3 <= 100)
):
    promedio = round(((num1 + num2 + num3) / 3), 2)
    if promedio < 51:
        mensaje = f"""Su nota es: {promedio} <br> Nota: Insuficiente"""
    elif promedio >= 51 and promedio <= 59:
        mensaje = f"""Su nota es: {promedio} <br> Nota: Suficiente"""
    elif promedio >= 60 and promedio <= 74:
        mensaje = f"""Su nota es: {promedio} <br> Nota: Bien"""
    elif promedio >= 75 and promedio <= 90:
        mensaje = f"""Su nota es: {promedio} <br> Nota: Notable"""
    elif promedio > 90:
        mensaje = f"""Su nota es: {promedio} <br> Nota: Sobresaliente"""
else:
    mensaje = f" Uno o varios datos no son correctos"


print(f"{mensaje}")

File: Tema 3 -Sentencia condicional (if-switch)/ejercicios/ejercicio8 nota ampliada/archives/8.php
<?php
$num1 = ($_POST["num1"]);
$num2 = ($_POST["num2"]);
$num3 = ($_POST["num3"]);
if (is_numeric($num1) && is_numeric($num2) && is_numeric($num3)) {
	$num1 = floatval($num1);
	$num2 = floatval($num2);
	$num3 = floatval($num3);
	if (($num1 >= 0 &&$num1 <= 100)&&($num2 >= 0 && $num2 <= 100) && ($num3 >= 0 && $num3 <= 100)) {
		$promedio = round((($num1 + $num2 + $num3) / 3), 2);
		if ($promedio < 51) {
			$mensaje = "Su nota es :  $promedio <br>Nota: Insuficiente";
		} else if ($promedio >= 51 && $promedio <= 59) {
			$mensaje = "Su nota es :  $promedio <br>Nota: Suficiente";
		} else if ($promedio >= 60 && $promedio <= 74) {
			$mensaje = "Su nota es :  $promedio <br>Nota: Bien";
		} else if ($promedio >= 75 && $promedio <= 90) {
			$mensaje = "Su nota es :  $promedio <br>Nota: Notable";
		} else if ($promedio > 90) {
			$mensaje = "Su nota es :  $promedio <br>Nota: Sobresaliente";
		}
	} else {
		$mensaje = " Uno varios datos no son validos";
	}
} else {
	$mensaje = "Ingrese datos en el formulario";
}
print $mensaje;
